### PR TITLE
Currency conversion on adapter JSON tests

### PR DESCRIPTION
**Pull request by guscarreon**
Pull request #1901 granted adapters the ability to convert currency inside their `MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo)` implementation. This pull request makes possible for adapters to include their own currency conversion rates in their JSON tests in order to better assert #1901's features

Also commited in this PR are JSON tests for the impactify adapter. Impactify's implementation of `MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo)` makes use of currency conversion and the additions to `adapters/impactify/impactifytest/exemplary/sample_banner.json` assert the correct conversion fo the impression bid floor, and on the other hand `adapters/impactify/impactifytest/supplemental/currency_rate_not_found.json` asserts a currency conversion error scenario.

@thomasdseao we'd like to hear your take on the tests added to your adapter. Please let us know if you approve.

### CODE REVIEW

**Review by VeronikaSolovei9 — APPROVED** [review]

**Comment by mansinahar** [comment]
@thomasdseao This PR makes some changes to the Impactify adpater's JSON tests to be able to test the currency conversion functionality being used in the `MakeRequests` implementation of Impactify. Can you please take a look when you have a chance and verify if the changes to those tests look good to you? Thanks!

File: adapters/adapterstest/test_json.go [hunk 1]
```diff
@@ -103,17 +105,49 @@ func loadFile(filename string) (*testSpec, error) {
 //
 // More assertions will almost certainly be added in the future, as bugs come up.
 func runSpec(t *testing.T, filename string, spec *testSpec, bidder adapters.Bidder, isAmpTest, isVideoTest bool) {
-	reqInfo := adapters.ExtraRequestInfo{}
+	reqInfo := getTestExtraRequestInfo(t, filename, spec, isAmpTest, isVideoTest)
+
```
Comment (mansinahar) [inline adapters/adapterstest/test_json.go:109]: Super nitpick: Please remove the new line

File: adapters/adapterstest/test_json.go [hunk 2]
```diff
@@ -103,17 +105,49 @@ func loadFile(filename string) (*testSpec, error) {
...
+	reqWrapper := &openrtb_ext.RequestWrapper{}
+	reqWrapper.BidRequest = &spec.BidRequest
+
+	reqExt, err := reqWrapper.GetRequestExt()
+	assert.NoError(t, err, "Could not unmarshall test request ext. %s", filename)
+
+	reqPrebid := reqExt.GetPrebid()
+	if reqPrebid != nil && reqPrebid.CurrencyConversions != nil && len(reqPrebid.CurrencyConversions.ConversionRates) > 0 {
+		err = currency.ValidateCustomRates(reqPrebid.CurrencyConversions)
+		assert.NoError(t, err, "Error unmarshalling test currency rates. %s", filename)
```
Comment (mansinahar) [inline adapters/adapterstest/test_json.go:131]: This doesn't have to be a unmarshalling error so please consider rewording this message to the following:
```
assert.NoError(t, err, "Error validating currency rates in the test request: %s", filename)
```

**Review by mansinahar — COMMENTED** [review]
Other than the two minor comments, this LGTM.

**Review by VeronikaSolovei9 — APPROVED** [review]

**Comment by thomasdseao** [comment]
@mansinahar Everything looks good 👍

**Review by mansinahar — APPROVED** [review]

**Comment by guscarreon** [comment]
> @mansinahar Everything looks good 👍

Thank you @thomasdseao

**Pull request merged**